### ISSUE TITLE
Time-Dependent Network configuration is not applied dynamically to agents on the links

### ISSUE THREAD

**Issue by Tim-Schwabe** [matsim-org/matsim-code-examples#1133]
We are trying to simulate some traffic on the highway by reducing speed limits of links for some limited time.

It seems that MATSim does not apply those speed limit changes to agents that are already on the link creating 2 problems:
* agents (already on the link) are not slowed down when freespeed gets limited
* agents (already on the link) do not speed up when freespeed is put back to normal

**Can this problem be solved with the XML input files or is this simply not meant to work our desired way?**

Input files:
```
<module name="network">
	<param name="inputNetworkFile" value="network.xml" />
	<param name="timeVariantNetwork" value="true" />
	<param name="inputChangeEventsFile" value="networkChangeEvents.xml" />
</module>
```
```
<networkChangeEvents>
	<networkChangeEvent startTime="05:09:00">
	    <link refId="38943"/>
	    <freespeed type="absolute" value="10.0"/>
	</networkChangeEvent>

	<networkChangeEvent startTime="05:14:00">
	    <link refId="38943"/>
	    <freespeed type="absolute" value="55.55"/>
	</networkChangeEvent>
</networkChangeEvents>
```

**Comment by JWJoubert** [comment]
@Tim-Schwabe maybe some comments/observations from my side. Remember that MATSim is not a micro-level traffic simulator. It was never meant to be one. Just keep that in mind.

It may be worth noting (or maybe you are aware, then just ignore this) that the agent's speed across a link is essentially determined when the agent _enters_ the link. I tried to explain the queue-based system better in [this short paper](http://dx.doi.org/10.1016/j.procs.2018.04.134).

One (easy?) way to maybe improve the granularity of your time problem could be to break your highway up into more segments. The idea (and its effect) is also what we tried explaining in the referenced paper.
